File: backend_node/src/controllers/cancion_album.controller.ts
import { Request, Response } from 'express'
import { execute_procedure } from '../configs/mysql.config';

export const insertarCancionAlbum = async(req: Request, res: Response) => {
    const { id_cancion, id_album } = req.body;
    
    try {
      //guardo al artista en la base de datos
      const insert_result = await execute_procedure('insertarCancionAlbum', {id_cancion, id_album});
      if (insert_result[0].respuesta == 0) return res.status(404).json({content: "La cancion con el id: " + id_cancion + ', no esta dispoible, porque ya se encuentra registrada para un album'});     
  
        
      //si todo sale correcto se devuelve un 200
      return res.status(200).json({
        content: "Se agrego correctamente el ID de la cancion: "+id_cancion+", para el ID del album: "+ id_album
      })
  
    } catch (error) {
      res.status(500).json({ content: 'Internal Error al registrar cancion, '+ error });
    }
  }

export const eliminarCancionAlbum = async(req: Request, res: Response) => {
  const { id_cancion, id_album } = req.body;

  try {
    const delete_result = await execute_procedure('eliminarCancionAlbum', {id_cancion, id_album});
    if (delete_result[0].respuesta == 0) return res.status(404).json({content: "Datos invalidos: ID cancion: " + id_cancion +', ID album: ,'+id_album+' porfavor verificar'});

    //si todo sale correcto se devuelve un 200
    return res.status(200).json({
    content:"Se ah elimino con exito la cancion: " + id_cancion+", para el ID del album: "+ id_album
    })
  } catch (error) {
    res.status(500).json({ content: 'Internal Error al elimnar cancion, '+ error });
  }
}

export const obtenerCancionesAlbum= async(req: Request, res: Response) => {
  const { id_album } = req.body;
  try {
    const get_canciones = await execute_procedure('obtenerCancionesAlbum', {id_album});
    if (get_canciones[0].respuesta == 0) return res.status(404).json({content: "No existe una album registrado con ID: " + id_album +', porfavor registrar previamente'});

    //si todo sale correcto se devuelve un 200
    return res.status(200).json({
      content: get_canciones
    })
  } catch (error) {
    res.status(500).json({ content: 'Internal Error al obtener canciones del album , '+ error });
  }
}

File: backend_python/src/controllers/cancion_album_controller.py
from decouple import config
from configs.mysql_config import execute_procedure
from configs.s3_config import uploadS3Base64

async def insertarCancionAlbum(data):
    try:

        # Guardo al artista en la base de datos
        insert_result = await execute_procedure('insertarCancionAlbum', [data['id_cancion'], data['id_album']])
        if insert_result.get('respuesta') == 0:
            return {'error404': 'La cancion con el id: ' + str(data['id_cancion']) + ', no esta dispoible, porque ya se encuentra registrada para un album'}

        new_insercion = {
            'content' : 'Se agrego correctamente el ID de la cancion' + str(data['id_cancion']) + ', para el ID del album' + str(data['id_album'])
        }
        return new_insercion

    except Exception as e:
        # Maneja errores aquí (por ejemplo, registro de errores o devolución de un mensaje de error)
        return {'error': str(e)}
    
async def eliminarCancionAlbum(data):
    try:
        delete_result = await execute_procedure('eliminarCancionAlbum', [data['id_cancion'], data['id_album']])
        if delete_result.get('respuesta') == 0:
            return {'error404': 'Datos invalidos: ID cancion: ' + str(data['id_cancion']) + ', ID album: ,' + str(data['id_album']) + ' porfavor verificar'}

        # Si todo sale correcto se devuelve un objeto artista para despues devolver un 200
        new_delete = {
            'content': 'Se elimino con exito la cancion: ' + str(data['id_cancion']) + ', para el ID del album: ' + str(data['id_album'])
        }
        return new_delete

    except Exception as e:
        # Maneja errores aquí (por ejemplo, registro de errores o devolución de un mensaje de error)
        return {'error': str(e)}

async def obtenerCancionesAlbum(data):
    try:
        get_canciones = await execute_procedure('obtenerCancionesAlbum', [data['id_album']])
        
          # Es una lista.
        if isinstance(get_canciones, list):
            new_album = {
                'content' : get_canciones
            }
            return new_album
        
        #Es un diccionario.
        if get_canciones.get('respuesta') == 0:
            return {'error404': 'No existe un album registrado con el ID: ' + str(data['id_album']) +', porfavor registrar previamente'}

        # Si todo sale correcto se devuelve un objeto artista para despues devolver un 200
        new_album = {
            'content' : [get_canciones]
        }
        return new_album

    except Exception as e:
        # Maneja errores aquí (por ejemplo, registro de errores o devolución de un mensaje de error)
        return {'error': str(e)}
